compiler/rustc_parse/src/parser/item.rs

```diff
@@ -997,35 +997,41 @@ impl<'a> Parser<'a> {
     fn parse_item_foreign_mod(
         &mut self,
         attrs: &mut Vec<Attribute>,
-        unsafety: Unsafe,
+        mut unsafety: Unsafe,
     ) -> PResult<'a, ItemInfo> {
         let sp_start = self.prev_token.span;
         let abi = self.parse_abi(); // ABI?
-        match self.parse_item_list(attrs, |p| p.parse_foreign_item(ForceCollect::No)) {
-            Ok(items) => {
-                let module = ast::ForeignMod { unsafety, abi, items };
-                Ok((Ident::empty(), ItemKind::ForeignMod(module)))
+        if self.token.is_keyword(kw::Unsafe) {
```

Comment:
Actually, now that I think of it, I'm not satisfied with the fact that this `if` block is larger than the rest of the function.. Do we expect `extern "C" unsafe` to show up much in practice?